### DISCUSSION TITLE
Provision tools 1.0.0

### DISCUSSION THREAD

**Discussion by Dadoum** [Dadoum/Provision#16] [Announcements]
First stable release of Provision !

Sideload IPA is not finished yet, so it is not included in this release.

<hr /><em>This discussion was created from the release <a href='https://github.com/Dadoum/Provision/releases/tag/1.0.0'>Provision tools 1.0.0</a>.</em>

**Comment by Dadoum** [comment]
On-going features (in order of priority):
 - No more libhybris, C++ and even C required!
 - 32 bit support
 - Cross-platform support
 - Sideload IPA rewrite (with Qt, libui, or anything lighter than GTK+ and cross-platform)

**Reply by roke-julian-lockhart** [reply]
@Dadoum, why no additional libhybris?

I suggest Qt, definitely. GTK+ is not easy-to-use cross-platform (and impossible without customization) and KDE Plasma is the current choice of desktop-environment for the Steam Deck, so it is gaining even more popularity very quickly.

**Reply by Dadoum** [reply]
Libhybris provides a lot of useful things in order to make a full Android emulator, but most of these things are useless for Provision (and it is logic, it was not the goal of libhybris). And since it adds the need for a C and a C++ compiler, I thought it would be cool if we could just load the 10 functions we need from the Apple libraries. 

Was thinking of Qt, bindings are one of the main issue with it but I think there are solutions.

**Reply by adityasanghvi** [reply]
Is it possible to add an armv7l anisette-server binary to the release?

**Reply by Dadoum** [reply]
@adityasanghvi unfortunately this release is not compatible with 32 bits architecture (and so incompatible with armv7). However, all the logic works on 32 bits, and I should port it at some time in the future (but I am very busy lately).

Otherwise, you can probably port it yourself. It's not that hard, the only trouble is that the thing I use for the server (which is named "archttp") does not support 32 bits, and so anisette-server can't. But if you use any other server, or just make altserver-linux use credentials directly from retrieve-headers (should be few lines of code), it will work.

**Reply by Dadoum** [reply]
https://github.com/NyaMisty/AltServer-Linux/blob/new/src/AnisetteDataManager.cpp#L60 here is the code that has to be replaced, remove http call and replace it with a process call in C++.

**Reply by adityasanghvi** [reply]
Thanks! I will try to spend time this weekend to work on this.

**Reply by adityasanghvi** [reply]
Reporting back on a successful implementation of your suggestion. Compiled retrieve_headers for armv7 and called it directly from AltServer function as you've outlined.

**Reply by Dadoum** [reply]
@adityasanghvi Don't hesitate to publish your edits of AltServer-Linux on GitHub. I know some people that would be interested, and I also added a build artefact for armv7, to help people to reproduce your setup (but unfortunately the build can only work on very recent OS).
